src/containers/artist-dashboard-page/WalletModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { useSelector } from 'utils/reducer'
+import { Nullable } from 'utils/typeUtils'
+import ClaimingModalBody from './components/ClaimingModalBody'
+import ClaimSuccessBody from './components/ClaimSuccessBody'
+import ReceiveBody from './components/ReceiveBody'
+import SendInputBody from './components/SendInputBody'
+import SendInputConfirmation from './components/SendInputConfirmation'
+import SendInputSuccess from './components/SendInputSuccess'
+import styles from './WalletModal.module.css'
+
+const messages = {
+  claimingTitle: "Hold Tight, We're Claiming Your $AUDIO!",
+  claimSuccess: 'Your $AUDIO Is Claimed!',
+  claimError: 'Uh oh! Something went wrong with your claim.',
+  receive: 'Receive $AUDIO',
+  send: 'Send $AUDIO',
+  confirmSend: 'Send $AUDIO',
+  sent: 'Your $AUDIO Has Been Sent',
+  sendError: 'Uh oh! Something went wrong sending your $AUDIO.'
+}
+
+const titlesMap = {
+  CLAIM: {
+    CLAIMING: messages.claimingTitle,
+    SUCCESS: messages.claimSuccess,
+    ERROR: messages.claimError
+  },
+  RECEIVE: {
+    KEY_DISPLAY: messages.receive
+  },
+  SEND: {
+    INPUT: messages.send,
+    AWAITING_CONFIRMATION: messages.confirmSend,
+    CONFIRMED_SEND: messages.sent,
+    ERROR: messages.sendError
+  }
+}
+
+const getTitle = (state: ModalState) => {
+  if (!state?.stage) return ''
+  switch (state.stage) {
+    case 'CLAIM':
+      return titlesMap.CLAIM[state.flowState.stage]
+    case 'RECEIVE':
+      return titlesMap.RECEIVE[state.flowState.stage]
+    case 'SEND':
+      return titlesMap.SEND[state.flowState.stage]
+  }
+}
+
+const ClaimErrorBody = () => null
+const SentErrorBody = () => null
+
+export const ModalBodyWrapper = ({
+  children
+}: {
+  children: React.ReactNode
+}) => {
+  return <div className={styles.modalContainer}>{children}</div>
+}
+
+type ModalContentProps = {
+  modalState: ModalState
+  onInputSendData: (amount: BN, wallet: WalletAddress) => void
+  onConfirmSend: () => void
+}
+
+const ModalContent = ({
+  modalState,
+  onInputSendData,
+  onConfirmSend
+}: ModalContentProps) => {
+  const balance = useSelector(getAccountBalance) ?? new BN('0')
+  const account = useSelector(getAccountUser)
+  const claimableBalance = useSelector(getClaimableBalance) ?? new BN('0')
+  const amountPendingTransfer = useSelector(getSendData)
+
+  if (!modalState || !account) return null
+
+  // This silly `ret` dance is to satisfy
+  // TS's no-fallthrough rule...
+  let ret: Nullable<JSX.Element> = null
+
+  switch (modalState.stage) {
+    case 'CLAIM': {
+      const claimStage = modalState.flowState.stage
+      switch (claimStage) {
+        case 'CLAIMING':
+          ret = <ClaimingModalBody balance={claimableBalance} />
+          break
+        case 'SUCCESS':
+          ret = <ClaimSuccessBody balance={balance} />
+          break
+        case 'ERROR':
+          ret = <ClaimErrorBody />
+          break
+      }
+      break
+    }
+    case 'RECEIVE': {
+      // @ts-ignore
```

Comment:
i dont think this ts ignore does anything since it applies to the next line ?

Reply:
TS is apparently smart enough to skip the comment - it breaks w/o this line